### PR TITLE
AF-22 improving error messages, including Welsh translations

### PR DESCRIPTION
**Pull request by russellthorn**

### CODE REVIEW

File: app/forms/GiveReasonFormProvider.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ class GiveReasonFormProvider @Inject() extends Mappings {
   def apply(): Form[GiveReasonQuestions] =
     Form(
       mapping(
-        "value"  -> optional(enumerable[GiveReason]()),
+        "value"  -> optional(enumerable[GiveReason](invalidKey = "GiveReason.error")),
```
Comment (sureshhmrc) [inline app/forms/GiveReasonFormProvider.scala:30]: please use lowercase `give....`  key to keep it consistant ?

File: app/forms/QuestionsFormProvider.scala [hunk 2]
```diff
@@ -30,12 +30,12 @@ class OtherQuestionsFormProvider @Inject() extends Mappings {
   def apply(): Form[OtherQuestions] =
     Form(
       mapping(
-        "ableToDo"     -> optional(enumerable[AbleToDo]()),
-        "howEasyScore" -> optional(enumerable[HowEasyQuestion]()),
+        "ableToDo"     -> optional(enumerable[AbleToDo](invalidKey = "ableToDo.error")),
+        "howEasyScore" -> optional(enumerable[HowEasyQuestion](invalidKey = "howEasy.error")),
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:34]: minor but better use field names for error prefixes like `howEasyScore.error` ?

File: app/forms/QuestionsFormProvider.scala [hunk 3]
```diff
@@ -30,12 +30,12 @@ class OtherQuestionsFormProvider @Inject() extends Mappings {
...
         "whyGiveScore" ->
           optional(text("whyGiveScore.error.required")
             .verifying(maxLength(maxFieldSizeWhyGiveScore, "whyGiveScore.error.maxlength"))),
-        "howDoYouFeelScore" -> optional(enumerable[HowDoYouFeelQuestion]())
+        "howDoYouFeelScore" -> optional(enumerable[HowDoYouFeelQuestion](invalidKey = "howFeelAboutService.error"))
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:38]: `howDoYouFeelScore.error` ?

File: app/forms/QuestionsFormProvider.scala [hunk 4]
```diff
@@ -50,12 +50,12 @@ class PTAQuestionsFormProvider @Inject() extends Mappings {
         "neededToDo" ->
           optional(text("neededToDo.error.required")
             .verifying(maxLength(maxFieldSizeAbleToDo, "neededToDo.error.maxlength"))),
-        "ableToDo"     -> optional(enumerable[AbleToDo]()),
-        "howEasyScore" -> optional(enumerable[HowEasyQuestion]()),
+        "ableToDo"     -> optional(enumerable[AbleToDo](invalidKey = "ableToDo.error")),
+        "howEasyScore" -> optional(enumerable[HowEasyQuestion](invalidKey = "howEasy.error")),
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:54]: same comment for all others

File: app/forms/QuestionsFormProvider.scala [hunk 5]
```diff
@@ -112,13 +112,13 @@ class PensionQuestionsFormProvider @Inject() extends Mappings {
...
         "whyGiveScore" ->
           optional(text("whyGiveScore.error.required")
             .verifying(maxLength(maxFieldSizeWhyGiveScore, "whyGiveScore.error.maxlength"))),
-        "howDoYouFeelScore" -> optional(enumerable[HowDoYouFeelQuestion]()),
-        "likelyToDo"        -> optional(enumerable[LikelyToDoQuestion]())
+        "howDoYouFeelScore" -> optional(enumerable[HowDoYouFeelQuestion](invalidKey = "howFeelAboutService.error")),
+        "likelyToDo"        -> optional(enumerable[LikelyToDoQuestion](invalidKey = "LikelyToDoQuestion.error"))
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:121]: lowercase `likelyToDo.error`

File: app/forms/QuestionsFormProvider.scala (hunk 2 above)
Comment (russellthorn) [reply app/forms/QuestionsFormProvider.scala:34]: not `howEasyQuestion` to match the enumerable?

Comment (russellthorn) [reply app/forms/QuestionsFormProvider.scala:34]: How much does it matter @sureshhmrc @struong ?

File: app/forms/QuestionsFormProvider.scala [hunk 6]
```diff
@@ -129,12 +129,15 @@ class CCGQuestionsFormProvider @Inject() extends Mappings {
   def apply(): Form[CCGQuestions] =
     Form(
       mapping(
-        "complianceCheckUnderstanding" -> optional(enumerable[CheckUnderstandingQuestion]()),
-        "treatedProfessionally"        -> optional(enumerable[TreatedProfessionallyQuestion]()),
+        "complianceCheckUnderstanding" -> optional(
+          enumerable[CheckUnderstandingQuestion](invalidKey = "complianceCheckUnderstanding.error")),
+        "treatedProfessionally" -> optional(
+          enumerable[TreatedProfessionallyQuestion](invalidKey = "complianceTreatedProfessionally.error")),
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:135]: this is just `treatedProfessionally.error` ?

File: app/forms/QuestionsFormProvider.scala [hunk 7]
```diff
@@ -145,13 +148,15 @@ class NmwCcgQuestionsFormProvider @Inject() extends Mappings {
   def apply(): Form[NmwCcgQuestions] =
     Form(
       mapping(
-        "treatedProfessionally" -> optional(enumerable[TreatedProfessionallyQuestion]()),
-        "checkUnderstanding"    -> optional(enumerable[CheckUnderstandingQuestion]()),
+        "treatedProfessionally" -> optional(
+          enumerable[TreatedProfessionallyQuestion](invalidKey = "treatedProfessionally.error")),
+        "checkUnderstanding" -> optional(
+          enumerable[CheckUnderstandingQuestion](invalidKey = "checkUnderstandingQuestion.error")),
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:154]: `checkUnderstanding.error`

File: app/forms/QuestionsFormProvider.scala [hunk 8]
```diff
@@ -145,13 +148,15 @@ class NmwCcgQuestionsFormProvider @Inject() extends Mappings {
...
         "whyGiveAnswer" ->
           optional(
             text("whyGiveAnswer.error.required")
               .verifying(maxLength(maxFieldSizeWhyGiveAnswer, "whyGiveAnswer.error.maxlength"))),
-        "supportFutureNmw" -> optional(enumerable[SupportFutureQuestion]())
+        "supportFutureNmw" -> optional(enumerable[SupportFutureQuestion](invalidKey = "supportFutureQuestion.error"))
```
Comment (sureshhmrc) [inline app/forms/QuestionsFormProvider.scala:159]: `supportFutureNmw.error`

File: app/forms/QuestionsFormProvider.scala (hunk 6 above)
Comment (sureshhmrc) [reply app/forms/QuestionsFormProvider.scala:135]: this is fine for now, as you already have another form that uses `treatedProfessionally` field name

File: app/forms/QuestionsFormProvider.scala (hunk 7 above)
Comment (russellthorn) [reply app/forms/QuestionsFormProvider.scala:154]: feel free to make these changes, like I make suggested changes to your pull requests.

**Review by struong — APPROVED** [review]

**Pull request merged**